src/mscorlib/src/System/Text/Encoding.cs

```diff
@@ -385,6 +384,11 @@ internal void SerializeEncoding(SerializationInfo info, StreamingContext context
             }
         }
 
+        // On Desktop, encoding instances that aren't cached in a static field are cached in
+        // a hash table by codepage.
+        private static volatile Hashtable encodings;
```

Comment:
Maybe this should be converted to a `Dictionary<int, Encoding>` instead to help reduce usages of the non-generic collections.